server/src/main.rs

```diff
@@ -5,9 +5,14 @@ use hover::get_hover_info;
 use jump_definition::get_definition_span;
 use shared_state::SharedState;
 use slicec::{compilation_state::CompilationState, slice_options::SliceOptions};
-use std::{path::PathBuf, sync::Arc};
+use std::{
+    collections::{HashMap, HashSet},
+    path::PathBuf,
+    sync::Arc,
+};
```

Comment:
I wonder if we can make the configuration a bit simpler if we define a struct that represents the configuration, for example:

```
// The extension configuration
pub struct SliceConfig {
    pub source_directory: Some(String),
    pub reference_directories: Some(Vector<String>),
}
```

Then we can read this struct from the settings, deserializing the JSON value:

```
let config = params
    .settings
    .get("slice")
    .ok_or(serde_json::Error::missing_field("slice"))
    .and_then(|value| Config::deserialize(value));
```

See https://github.com/johalun/rustkpi/blob/fbc4cab7e3caeab461f4ea822a184c104703ac0b/rust/rust-src-1.25.0/src/tools/rls/src/actions/notifications.rs#L141

Reply:
I agree it would be simpler and cleaner to group all the config into a single struct like you suggest.

I'm not sure if directly deserializing the struct would always be correct though, since sometimes (it seems like) we only get or care about subsets. But maybe this could be fixed with some refactoring.